appserver/extras/docker-images/micro/src/main/docker/Dockerfile

```diff
@@ -69,5 +72,5 @@ WORKDIR ${HOME_DIR}
 COPY --chown=payara:payara maven/bin/* ${SCRIPT_DIR}/
 COPY --chown=payara:payara maven/artifacts/payara-micro.jar .
 
-ENTRYPOINT ["/bin/sh", "entrypoint.sh"]
-CMD ["--deploymentDir","/opt/payara/deployments"]
+ENTRYPOINT ["tini", "--", "/opt/payara/entrypoint.sh"]
```

Comment:
I'm thinking why we arent using "shell form" of the command, which basically does the same thing that entrypoint does, just under the covers and hopefully with correct signal handling as well
```suggestion
ENTRYPOINT java -XX:MaxRAMPercentage=${MEM_MAX_RAM_PERCENTAGE} -Xss${MEM_XSS} -XX:+UseContainerSupport ${JVM_ARGS} -jar payara-micro.jar
```

can you try if this would be sufficient? You can drop tini and the entrypoint file altogether then.

Reply:
I tried and it works! The shutdown hook is called in micro